packages/cli/lib/lib/webpack/webpack-sw-config.js

```diff
@@ -0,0 +1,62 @@
+const webpack = require('webpack');
+const merge = require('webpack-merge');
+const baseConfig = require('./webpack-base-config');
+const BabelEsmPlugin = require('babel-esm-plugin');
+const fs = require('fs');
+const {yellow} = require('chalk');
+const { resolve } = require('path');
+
+function swConfig(config) {
+  const { dest, src, brotli, esm } = config;
+
+  const plugins = [
+    new webpack.DefinePlugin({
+      'process.env.ES_BUILD': false,
+      'process.env.ENABLE_BROTLI': brotli,
+      'process.env.NODE_ENV': 'production',
+    }),
+  ];
+
+  esm && plugins.push(
+    new BabelEsmPlugin({
+      filename: '[name]-esm.js',
+      beforeStartExecution: (plugins) => {
+        plugins.forEach(plugin => {
+          if (plugin.constructor.name === 'DefinePlugin' && plugin.definitions) {
+            for (const definition in plugin.definitions) {
+              if (definition === 'process.env.ES_BUILD') {
+                plugin.definitions[definition] = true;
+              }
+            }
+          } else if (plugin.constructor.name === 'DefinePlugin' && !plugin.definitions) {
+            throw new Error('WebpackDefinePlugin found but not `process.env.ES_BUILD`.');
+          }
+        });
+      }
+    })
+  );
+
+  let swSrc = resolve(__dirname, './../sw.js');
+  if (fs.existsSync(resolve(`${src}/sw.js`))) {
+    console.log(yellow('⚛️ Info - CLI found sw.js in source root folder, using that to compile the final service worker instead'));
```

Comment:
Nit: can we make this a different color like blue? yellow seems like a warning, this is informational.

Also maybe shorten the message since it's a heads-up that will be shown every build? Or only show the message when things change (first build or sw.js added/removed):

```js
  // only warn on first build, OR on rebuild if sw.js is added/removed:
  let swSrc = resolve(__dirname, './../sw.js');
  const exists = fs.existsSync(resolve(`${src}/sw.js`));
  if (exists !== global.hasSeenSw) {
    global.hasSeenSw = exists;
    if (exists) {
      console.log(blue('⚛️ Detected custom sw.js: compiling instead of default Service Worker.'));
    }
    else {
      console.log(blue('⚛️ No custom sw.js detected: compiling default Service Worker.'));
    }
  }
```

Reply:
done